examples/indentation/indentation_ast.go

```diff
@@ -0,0 +1,247 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+)
+
+func main() {
+	in := os.Stdin
+	if len(os.Args) > 1 {
+		f, err := os.Open(os.Args[1])
+		if err != nil {
+			log.Fatal(err)
+		}
+		defer f.Close()
+		in = f
+	}
+	pn, err := ParseReader("", in)
+	if err != nil {
+		log.Fatal(err)
+	}
+	ret, err := pn.(ProgramNode).exec()
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Println(ret)
+}
+
+var lvalues = make(map[string]int)
+
+// element of statements
```

Comment:
I suggest to use unexported types here, as this is only example code. Otherwise `make lint` complains about the exported types not being commented properly:

```
examples/indentation/indentation_ast.go:34:1: comment on exported type Executor should be of the form "Executor ..." (with optional leading article)
```

This is true for all the exported types in this example.

Reply:
I just expand all comments